src/codegen/sdk/core/expressions/chained_attribute.py

```diff
@@ -74,6 +76,41 @@ def attribute(self) -> Attribute:
         """
         return self._attribute
 
+    @property
+    @reader
+    def attribute_chain(self) -> list["FunctionCall | Name"]:
+        from codegen.sdk.core.detached_symbols.function_call import FunctionCall
+
+        ret = []
+        curr = self
+
+        # Traverse backwards in code (children of tree node)
+        while isinstance(curr, ChainedAttribute):
```

Comment:
Lots of ininstance checks like these indicate this should probably be done via OOP - one method on chained attribute and one on function call.